packages/vm/tests/api/EIPs/eip-3074-authcall.spec.ts

```diff
@@ -0,0 +1,680 @@
+import tape from 'tape'
+import {
+  Address,
+  bigIntToBuffer,
+  bufferToBigInt,
+  ecsign,
+  keccak256,
+  privateToAddress,
+  setLengthLeft,
+  toBuffer,
+  zeros,
+} from 'ethereumjs-util'
+import Common, { Chain, Hardfork } from '@ethereumjs/common'
+import VM from '../../../src'
+import { Transaction } from '@ethereumjs/tx'
+import { Block } from '@ethereumjs/block'
+import { ERROR } from '../../../src/exceptions'
+import { InterpreterStep } from '../../../src/evm/interpreter'
+
+const common = new Common({
+  chain: Chain.Mainnet,
+  hardfork: Hardfork.London,
+  eips: [3074],
+})
+
+// setup the accounts for this test
+const privateKey = Buffer.from(
+  'e331b6d69882b4cb4ea581d88e0b604039a3de5967688d3dcffdd2270c0fd109',
+  'hex'
+)
+
+const block = Block.fromBlockData(
+  {
+    header: {
+      baseFeePerGas: BigInt(7),
+    },
+  },
+  { common }
+)
+
+const callerPrivateKey = Buffer.from('44'.repeat(32), 'hex')
+const callerAddress = new Address(privateToAddress(callerPrivateKey))
+const PREBALANCE = BigInt(10000000)
+
+const address = new Address(privateToAddress(privateKey))
+const contractAddress = new Address(Buffer.from('ff'.repeat(20), 'hex'))
+const contractStorageAddress = new Address(Buffer.from('ee'.repeat(20), 'hex'))
+
+// Bytecode to exit call frame and return the topmost stack item
+const RETURNTOP = Buffer.from('60005260206000F3', 'hex')
+// Bytecode to store CALLER in slot 0 and GAS in slot 1 and the first 32 bytes of the input in slot 2
+// Returns the entire input as output
+const STORECALLER = Buffer.from('5A60015533600055600035600255366000600037366000F3', 'hex')
+
+/**
+ * This signs a message to be used for AUTH opcodes
+ * @param commitUnpadded - The commit which we want to sign
+ * @param address - The contract address we are using AUTH on
+ * @param privateKey - The private key of the account to sign
+ * @returns - Returns the signed message
```

Comment:
nit, no need for extra dash (same with all the params, but it doesn't seem VS Code adds an extra `-` like in this case)

> <img width="843" alt="Screen Shot 2022-03-24 at 12 19 26 PM" src="https://user-images.githubusercontent.com/22116/159993890-f91ac735-987c-4f32-b3f1-edb565e24e49.png">

```suggestion
 * @returns The signed message
```